src/Symfony/Component/VarDumper/Test/VarDumperTestTrait.php

```diff
@@ -31,15 +48,22 @@ public function assertDumpMatchesFormat($expected, $data, $filter = 0, $message
 
     protected function getDump($data, $key = null, $filter = 0)
     {
-        $flags = getenv('DUMP_LIGHT_ARRAY') ? CliDumper::DUMP_LIGHT_ARRAY : 0;
-        $flags |= getenv('DUMP_STRING_LENGTH') ? CliDumper::DUMP_STRING_LENGTH : 0;
-        $flags |= getenv('DUMP_COMMA_SEPARATOR') ? CliDumper::DUMP_COMMA_SEPARATOR : 0;
+        $flags = self::$flags;
+        if (null === $flags) {
+            $flags = getenv('DUMP_LIGHT_ARRAY') ? CliDumper::DUMP_LIGHT_ARRAY : 0;
+            $flags |= getenv('DUMP_STRING_LENGTH') ? CliDumper::DUMP_STRING_LENGTH : 0;
+            $flags |= getenv('DUMP_COMMA_SEPARATOR') ? CliDumper::DUMP_COMMA_SEPARATOR : 0;
+        }
+
+        if (!self::$varCloner) {
```

Comment:
`null === self::$varCloner`?

Reply:
Would be my own preference too, but I think (might be wrong) the CS in core is this one when there is no ambiguity on the possible types inside `self::$varCloner` (i.e that's either `null` or a `VarCloner` instance. No bool casting ambiguity is possible).

Reply:
I prefer `if (!self::$varCloner) {` :)